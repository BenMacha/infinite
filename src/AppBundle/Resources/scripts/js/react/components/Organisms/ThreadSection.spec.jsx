import React from 'react';
// eslint-disable-next-line import/no-extraneous-dependencies
import renderer from 'react-test-renderer';

import ThreadSection from './ThreadSection.jsx';

describe('ThreadSection container', () => {
  const header = {
    text: 'Header text',
    tag: 'h1',
  };

  it('render correctly', () => {
    const articles = [
      {
        id: 42,
        publication_url: '/path/to/article/42',
        publish_date: '2016-08-31T12:54:52.956Z',
        content: {
          restricted: false,
          title: 'Article without type',
        },
      },
      {
        id: 42,
        publication_url: '/path/to/article/42',
        publish_date: '2016-08-31T12:54:52.956Z',
        content: {
          restricted: false,
          title: 'The article title',
        },
        type: 'article',
      },
      {
        id: 42,
        publication_url: '/path/to/alert/42',
        publish_date: '2016-08-31T12:54:52.956Z',
        content: {
          text: 'The article text',
          title: 'The article title',
        },
        type: 'alert',
      },
      {
        type: 'advert',
        id: 'my-advert-id',
      },
    ];
    const tree = renderer.create(
      <ThreadSection
        header={header}
        articles={articles}
        qaId="test-id"
      />
    );

    expect(tree).toMatchSnapshot();
  });

  describe('with more button', () => {
    it('render correctly', () => {
      const tree = renderer.create(
        <ThreadSection
          header={header}
          more
          qaId="test-id"
        />
      );

      expect(tree).toMatchSnapshot();
    });

    it('calls callback on button click', () => {
      const onButtonClick = jest.fn();
      const tree = renderer.create(
        <ThreadSection
          header={header}
          more
          onButtonClick={onButtonClick}
          qaId="test-id"
        />
      );

      const button = tree.toJSON().children[1].children[0];

      button.props.onClick();

      expect(onButtonClick).toBeCalled();
    });
  });
});
