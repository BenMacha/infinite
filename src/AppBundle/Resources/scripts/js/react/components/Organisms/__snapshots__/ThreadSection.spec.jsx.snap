exports[`ThreadSection container render correctly 1`] = `
<div>
  <ul
    className="cm-section-thread__section-list">
    <li
      className="cm-section-thread__article" />
    <li
      className="cm-section-thread__article" />
    <li
      className="cm-section-thread__article" />
    <li
      className="cm-section-thread__article" />
  </ul>
</div>
`;

exports[`ThreadSection container with more button render correctly 1`] = `
<div>
  <ul
    className="cm-section-thread__section-list" />
  <div
    className="ol-grid__cell ol-grid__cell--full@to-phone-l">
    <a
      aria-label="Afficher les contenus plus anciens"
      className="ca-btn ca-btn--important"
      onClick={[Function]}>
      Voir Plus
    </a>
  </div>
</div>
`;
